### DISCUSSION TITLE
classes/class_area2d

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#41] [User-contributed notes]
# classes/class_area2d

Inherits: CollisionObject2D< Node2D< CanvasItem< Node< Object A region of 2D space that detects other CollisionObject2D s entering or exiting it. Description: Area2D is a region of 2D space defined...

https://docs.godotengine.org/en/4.3/classes/class_area2d.html

**Comment by LenaKotik** [comment]
why does Area2D need an AudioBus?

**Reply by Calinou** [reply]
The AudioBus properties are used to apply automatic effects when the camera is inside (typically a reverb effect in indoor areas). They are optional – you can leave them empty if you don't need this feature.

**Comment by FoxyFox909** [comment]
When using Area2Ds that are in high speed movement (example: an Area2D to define some room inside a ship that's moving at high speed), and you have a CharacterBody2D that you want to collide with the area, you might have to use a RigidBody2D as a proxy hitbox due to the updates of CharacterBody2D and Area2D collisions having an offset cause by the movement (the CollisionShapes will look like they should enable a collision, but the actual place at which a collision occurs will be offset).

However, a RigidBody2D fixes that. Its collisions with an Area2D will be accurately reported even at high speeds. You might need to turn on Continuous Collision Detection on it.

**Comment by KestrelDotNet** [comment]
How to use Signals like area_entered in C#. I tried to use this using C# script but not working but using manual works.
Example scene structure:
1. Node2D - parent
1.1. Area2D - Name = BoxA - moving left and right, collide to BoxB
1.1.1. CollisionShape2D
1.2. Area2D - Name = BoxB - stationary in center. (No script)
1.2.1. CollisionnShape2D
script of BoxA
```csharp
public partial class BoxA : Area2D
{
    private Action<Area2D> handleSignalAreaEntered;
    public override void _Ready()  { 
          handleSignalAreaEntered = OnAreaEnteredBoxA;       
          this.Connect(BoxB.SignalName.AreaEntered, Callable.From(handleSignalAreaEntered));
    }
    public override void _Process(double delta)
    {
          // logic  moving left and right
    }
    private void OnAreaEnteredBoxA(Area2D area) {
        GD.Print($"{Name} receive signal emit by {area.Name}"); 
    }
}
```
Method OnAreaEnteredBoxA doesnt work on Connect but when I try manual connection the method OnAreaEnteredBoxA triggers.

**Comment by NancokPS2** [comment]
Area2D seems to need monitoreable = true in order to detect TileMapLayers. This doesn't seem to be noted anywhere.

**Reply by TheBen27** [reply]
This also seems to be true for static bodies, but not rigid bodies.

**Comment by w-dib** [comment]
Is there any instance at all where `overlaps_area(area)` is better than using `on_area_entered` signal?

**Reply by NancokPS2** [reply]
When you want to check all current entered areas instead of 1 at a time.

**Comment by RobertMic** [comment]
There seems to currently be [a bug](https://github.com/godotengine/godot/issues/81883) (at least in godot 4.5.1) where if a `StaticBody2D` moves inside of an `Area2D`, it will not be detected by the `Area2D`. From my testing, it also seems that a `RigidBody2D` with `freeze = true` has the same bugs. There are some caveats to this:
- If a scene is first loaded and a `StaticBody2D` is inside of an `Area2D` it **will** be detected
- If the `Area2D` moves as well, then it will detect any `StaticBody2D`s inside of it

The workaround seems to be to use an `AnimatableBody2D` instead of a `StaticBody2D`. I've also had some success with using a `RigidBody2D` with `sleeping = true` and `gravity = 0`
